falcon/testing/base.py

```diff
@@ -47,13 +47,13 @@ class TestBase(unittest.TestCase):
             can use to add a route to the API.
     """
 
-    def setUp(self):
+    def setUp(self, api=None, srmock=None):
         """Initializer, unittest-style"""
 
         super(TestBase, self).setUp()
         self._id = itertools.count(0)
-        self.api = falcon.API()
-        self.srmock = StartResponseMock()
+        self.api = api or falcon.API()
```

Comment:
Nitpick: It's a good habit to explicitly test for None (e.g., `falcon.API() if api is None else api`) since the object being passed in may (unexpectedly) evaluate to False, but in this case I don't think it matters since we are expecting instances of API and StartResponseMock which are known to always evaluate to True.